stories/molecules/codeExpression/codeExpression.stories.js

```diff
@@ -0,0 +1,28 @@
+import { storiesOf } from '@storybook/html'
+
+storiesOf('Molecules/CodeExpression', module)
+  .add(
+    'CodeExpression',
+    () => `<div>
+      <stencila-codeexpression
+        data-collapsed="false"
+        itemtype="stencila:CodeChunk"
+      >
+        <pre slot="text"><code>x * y</code></pre>
+        <pre slot="output"><output>Reaaaaalllyyyyyyyy loooongggggggggggggggggggg output</output></pre>
+      </stencila-codeexpression>
+    `
+  )
+  .add(
+    'CodeExpression within a paragraph',
+    () => `<p>
+      This is a paragraph with a code expresssion inside it
+      <stencila-codeexpression
+        data-collapsed="false"
+        itemtype="stencila:CodeChunk"
+      >
+        <pre slot="text"><code>x * y</code></pre>
+        <pre slot="output"><output>42</output></pre>
+      </stencila-codeexpression> followed by some more text.
+    </p>`
+  )
```

Comment:
I added this story because a `CodeExpression` is `InlineContent` and most (all?) of the time will be surrounded by text. Given that, I'm not sure that deriving from `CodeChunk` is the best option. I envisaged that this component would be more like a hovercard that would (eventually, when in editing mode) allow changing of `text` and `programmingLanguage`.

![image](https://user-images.githubusercontent.com/1152336/64898886-074b4280-d6dd-11e9-9eff-87d8ff8a8362.png)

Reply:
I'd like to spend some time to discuss the Encoda-ing of this component, and how best to handle progressive enhancement in such cases.
As far as I can tell, given the above schema Encoda will generate an invalid/unsemantic HTML structure. This is because the `p(aragraph)` will contain a `pre` element with the `output` of the `CodeExpression`, which isn't allowed.

We could have something similar to the below snippet:
```html
<stencila-code-expression language="perl"><code>x * y</code><output>42</output></stencila-code-expression>
```

The question then is if such code expressions are constrained to only allow one line expressions, and if we add heuristics to use `CodeChunk` for multi-line code expressions.
Another thing to note is that we'll lose the syntax-highlighting provided by PrismJS, as that only works with `pre>code` out of the box as far as I'm aware. We could explore using custom functions to tokenize the expression, and insert the resulting markup in the `code` block.

Re. editing: For now editing might be easiest and most intuitive to do inline, while we work on modals or drawers (side or bottom) for a more dedicated editing experience similar to Excel/Sheets function editor maybe.